---
title: "Hegde et al - 2025 - Myelopoiesis Epigenetics"
title-block-banner: true
subtitle: "Mouse Bone Marrow (BM) KP-Naive scRNA-seq analysis"
format: html
editor: visual
---

## Load libraries

```{r}
library(tidyverse)
library(Seurat)
library(SCP)
library(ComplexHeatmap)
library(colorspace)
library(skimr)
library(reactable)
library(DT)

set.seed(729)
```

## Load data

```{r, warning=FALSE}
# h5_files <- list.files("data/", pattern = "sample_filtered_feature_bc_matrix.h5", recursive = TRUE, full.names = TRUE)
# 
# sc.list <- lapply(h5_files, function(file) {
#   expr_matrix <- Read10X_h5(file)$`Gene Expression`
#   CreateSeuratObject(counts = expr_matrix, project = tools::file_path_sans_ext(basename(file)))
# })
# 
# names(sc.list) <- basename(dirname(dirname(h5_files))) 
```

```{r}
#sc.list
```

## Seurat

```{r}
# sc.list <- lapply(X = sc.list, FUN = NormalizeData) 
# sc.list <- lapply(X = sc.list, FUN = ScaleData)
# sc.list <- lapply(X = sc.list, FUN = SCTransform, method = "glmGamPoi")
# 
# features <- SelectIntegrationFeatures(object.list = sc.list, nfeatures = 3000)
# sc.list <- PrepSCTIntegration(object.list = sc.list, anchor.features = features)
# 
# sc.list <- lapply(X = sc.list, FUN = RunPCA, features = features)
# 
# immune.anchors <- FindIntegrationAnchors(object.list = sc.list, normalization.method = "SCT", anchor.features = features, dims = 1:30, reduction = "rpca", k.anchor = 20)
# 
# BM_object_combined_var <- IntegrateData(anchorset = immune.anchors, normalization.method = "SCT", dims = 1:30)
# BM_object_combined_var <- RunPCA(BM_object_combined_var, npcs = 30, verbose = FALSE) %>% 
#                           RunUMAP(reduction = "pca", dims = 1:30, verbose = FALSE) %>% 
#                           FindNeighbors(reduction = "pca", dims = 1:30, verbose = FALSE) %>%
#                           FindClusters(resolution = 0.7, verbose = FALSE)
```

## Annotation

```{r}
# sc_obj$SH_annots <- plyr::mapvalues(
#   x = sc_obj$seurat_clusters, 
#   from = c("0","1","2","3","4","5","6","7","8","9","10",
#            "11","12","13","14","15","16","17","18","19","20",
#            "21","22","23","24","25","26","27","28"), 
#   to = c(
#     'immature neu', 
#     'pre-neu_cycling', 
#     'pre-neu', 
#     'immature neu', 
#     'pre-mono', 
#     'Ly6Chi mono', 
#     'mature neu', 
#     'remove', 
#     'cMoP_cycling', 
#     'remove', 
#     'GMP', 
#     'mature neu', 
#     'preNeu', 
#     'cMoP', 
#     'GP', 
#     'remove', 
#     'preNeu_cycling', 
#     'remove', # DC-like
#     'Ly6Chi mono', 
#     'Ly6Clow mono', 
#     'remove', 
#     'MEP', 
#     'GMP', 
#     'pre-mono', 
#     'remove', 
#     'Ly6Chi proDC3', 
#     'remove', 
#     'remove', 
#     'remove' 
#   )
# )
# sc_obj = subset(sc_obj, annots != "remove")
# Idents(sc_obj) = "SH_annots"
# saveRDS(sc_obj, file = "Hegde_2025_Mouse_BM_KP_scRNA.rds")
```

```{r}
# SCP::CellDimPlot(sc_obj, group.by = "SH_annots", reduction = "umap", pt.size = 0.75, label = T, label_repel = T, label_insitu = T)
```

```{r, fig.width=15, fig.height=10}
# SCP::CellDimPlot(sc_obj, group.by = "SH_annots", split.by = "sample_simplified", reduction = "umap", pt.size = 0.75, label = T, label_repel = T, label_insitu = T, legend.position = "bottom")
```

## Reproduce Manuscript Figures

```{r}
sc_obj = readRDS("Hegde_2025_Mouse_BM_KP_scRNA.rds")
DefaultAssay(sc_obj) = "RNA"
```

## Export

```{r}
bc_to_annot_df = sc_obj@meta.data |> rownames_to_column("bc") |> dplyr::select(bc, seurat_clusters, SH_annots)
write.csv(bc_to_annot_df, file = "Hegde_2025_Mouse_BM_KP_scRNA_bc_cluster_annot.csv", row.names = F)
```

```{r}
table(sc_obj$sample, sc_obj$SH_annots)
```

```{r}
SCP::CellDimPlot(sc_obj, group.by = "SH_annots", reduction = "umap", pt.size = 0.75, label = T, label_repel = T, label_insitu = T)
```

```{r, fig.width=15, fig.height=10}
SCP::CellDimPlot(sc_obj, group.by = "SH_annots", split.by = "sample_simplified", reduction = "umap", pt.size = 0.75, label = T, label_repel = T, label_insitu = T, legend.position = "bottom")
```

## Markers

```{r}
degs_markers = FindAllMarkers(sc_obj, only.pos = T, logfc.threshold = 1, assay = "RNA")
datatable(degs_markers, filter = "top")
```

## Truth-plot

This code randomly selects cell IDs from each cell type in the manual_annotation vector. Hereâ€™s a step-by-step breakdown:

1.  **Reproducibility:**\
    `set.seed(123)` ensures that the random sampling is reproducible every time the code is run.

2.  **Iterating Over Cell Types:**\
    `unique(manual_annotation)` extracts each distinct cell type from the manual_annotation vector (such as "AM", "momac_Arg1", etc.).

3.  **Sampling Cell IDs:**\
    For each cell type, the function `sample(names(manual_annotation)[manual_annotation==v], min(sum(manual_annotation==v), 50))` randomly selects cell IDs (which are the names of the vector elements) corresponding to that cell type. The number of cells selected is the minimum between the total available for that type and 50. This means that if a cell type has more than 50 cells, only 50 are chosen; if it has fewer, all available cells are selected.

4.  **Combining Results:**\
    The sampled cell IDs from each cell type are combined into a single vector called `cells` using `as.vector(sapply(...))`.

In summary, the code creates a reproducible, random subsample of up to 50 cell IDs for each cell type present in the manual_annotation vector.

```{r}
markers_annots = readxl::read_xlsx("BM_markers.xlsx")
```

```{r}
mat <- sc_obj@assays$RNA$counts[unique(markers_annots$gene), ]
mat = mat[unique(markers_annots$gene), ]
manual_annotation = sc_obj$SH_annots
```

```{r}
reactable(markers_annots)
```

```{r}
set.seed(123)
cells = as.vector(sapply(unique(manual_annotation), function(v) sample(names(manual_annotation)[manual_annotation==v], min(sum(manual_annotation==v), 50))))
mat = mat[, unlist(cells)]
manual_annotation = manual_annotation[unlist(cells)]

dim(mat)
colnames(mat) = NULL
```

```{r, fig.width=15, fig.height=10}
unique_clusters = sort(unique(manual_annotation))
sample_color = qualitative_hcl(n = length(unique_clusters), palette = 'Dark 2')
names(sample_color) = unique_clusters
color = circlize::colorRamp2(breaks = c(0,1,2,3,5,10,20), colors = c('white', rev(sequential_hcl(n = 5,palette = 'OrRd')), 'black'))

gene_groups = markers_annots$group[match(rownames(mat), markers_annots$gene)]
colnames(mat) = NULL
h1 = Heatmap(t(as.matrix(mat)), row_names_side = "right", col = color,
             row_split = manual_annotation, show_row_dend = F,
             row_title_rot = 0, column_title_gp = gpar(fontsize = 12, fontface = "italic"),
             row_title_gp = gpar(fontsize = 15), cluster_columns = FALSE,cluster_rows = T,
             row_gap = unit(1, "mm"), column_gap = unit(1, "mm"), border = 'gray50',
             column_names_gp = gpar(fontface = "italic"),
             column_title_rot = 90,
             heatmap_legend_param = list(title = "Expression")
)

print(h1)
```

## Gene Signature Scoring with UCell

```{r}
library(UCell)
```

```{r}
markers_df <- read.csv("ucell_signatures.csv")
markers_list <- list(oxid_stress_response = markers_df$gene_name)
reactable(markers_df)
```

```{r, fig.width=8, fig.height=6}
sc_obj <- AddModuleScore_UCell(
  sc_obj, 
  features = markers_list)
signature.names <- paste0(names(markers_list), "_UCell")

VlnPlot(sc_obj,
        cols = c("#f44336", "#bcbcbc"), 
        signature.names, 
        split.by = "sample_simplified", pt.size=0, split.plot =T, sort = F)
```

## Session Info

```{r}
sessionInfo()
```
