<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Hegde et al - 2025 - Myelopoiesis Epigenetics</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/clipboard/clipboard.min.js"></script>
<script src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/quarto-html/quarto.js"></script>
<script src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/quarto-html/popper.min.js"></script>
<script src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/quarto-html/anchor.min.js"></script>
<link href="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/quarto-html/quarto-syntax-highlighting-07ba0ad10f5680c660e360ac31d2f3b6.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/libs/bootstrap/bootstrap-fe6593aca1dacbc749dc3d2ba78c8639.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Hegde et al - 2025 - Myelopoiesis Epigenetics</h1>
            <p class="subtitle lead">Mouse Bone Marrow (KP; Nfe2l2) scATAC-seq analysis</p>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content quarto-banner-title-block" id="quarto-document-content">





<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject
Loading required package: sp

Attaching package: 'SeuratObject'

The following objects are masked from 'package:base':

    intersect, t</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SeuratExtend)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratExtendData</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Signac)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scDblFinder)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment
Loading required package: SummarizedExperiment
Loading required package: MatrixGenerics
Loading required package: matrixStats

Attaching package: 'matrixStats'

The following object is masked from 'package:dplyr':

    count


Attaching package: 'MatrixGenerics'

The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars

Loading required package: GenomicRanges
Loading required package: stats4
Loading required package: BiocGenerics

Attaching package: 'BiocGenerics'

The following object is masked from 'package:SeuratObject':

    intersect

The following objects are masked from 'package:lubridate':

    intersect, setdiff, union

The following objects are masked from 'package:dplyr':

    combine, intersect, setdiff, union

The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs

The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,
    table, tapply, union, unique, unsplit, which.max, which.min

Loading required package: S4Vectors

Attaching package: 'S4Vectors'

The following objects are masked from 'package:lubridate':

    second, second&lt;-

The following objects are masked from 'package:dplyr':

    first, rename

The following object is masked from 'package:tidyr':

    expand

The following object is masked from 'package:utils':

    findMatches

The following objects are masked from 'package:base':

    expand.grid, I, unname

Loading required package: IRanges

Attaching package: 'IRanges'

The following object is masked from 'package:sp':

    %over%

The following object is masked from 'package:lubridate':

    %within%

The following objects are masked from 'package:dplyr':

    collapse, desc, slice

The following object is masked from 'package:purrr':

    reduce

Loading required package: GenomeInfoDb
Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.


Attaching package: 'Biobase'

The following object is masked from 'package:MatrixGenerics':

    rowMedians

The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians


Attaching package: 'SummarizedExperiment'

The following object is masked from 'package:Seurat':

    Assays

The following object is masked from 'package:SeuratObject':

    Assays</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plyranges)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'plyranges'

The following object is masked from 'package:IRanges':

    slice

The following objects are masked from 'package:dplyr':

    between, n, n_distinct

The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(JASPAR2020)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TFBSTools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code></code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnsDb.Mmusculus.v79)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ensembldb
Loading required package: GenomicFeatures
Loading required package: AnnotationDbi

Attaching package: 'AnnotationDbi'

The following object is masked from 'package:plyranges':

    select

The following object is masked from 'package:dplyr':

    select

Loading required package: AnnotationFilter

Attaching package: 'ensembldb'

The following object is masked from 'package:plyranges':

    filter

The following object is masked from 'package:dplyr':

    filter

The following object is masked from 'package:stats':

    filter</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BSgenome.Mmusculus.UCSC.mm10)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BSgenome
Loading required package: Biostrings
Loading required package: XVector

Attaching package: 'XVector'

The following object is masked from 'package:purrr':

    compact


Attaching package: 'Biostrings'

The following object is masked from 'package:base':

    strsplit

Loading required package: BiocIO

Attaching package: 'BiocIO'

The following object is masked from 'package:rtracklayer':

    FileForFormat</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>annotations <span class="ot">=</span> <span class="fu">GetGRangesFromEnsDb</span>(<span class="at">ensdb =</span> EnsDb.Mmusculus.v79)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)
Warning in .merge_two_Seqinfo_objects(x, y): The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(annotations) <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">'chr'</span>, <span class="fu">seqlevels</span>(annotations))</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">genome</span>(annotations) <span class="ot">=</span> <span class="st">"mm10"</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="st">'%notin%'</span> <span class="ot">&lt;-</span> <span class="cf">function</span>(x,y)<span class="sc">!</span>(<span class="st">'%in%'</span>(x,y))</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>start_time <span class="ot">=</span> <span class="fu">Sys.time</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-data" class="level2">
<h2 class="anchored" data-anchor-id="load-data">Load data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>sampleID <span class="ot">=</span> <span class="st">"BM_WT"</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>counts_mat <span class="ot">=</span> <span class="fu">Read10X_h5</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(<span class="st">"data/"</span>, sampleID, <span class="st">"/filtered_peak_bc_matrix.h5"</span>))</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"data/"</span>, sampleID, <span class="st">"/singlecell.csv"</span>), </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>chrom_assay <span class="ot">=</span> <span class="fu">CreateChromatinAssay</span>(<span class="at">counts =</span> counts_mat, <span class="at">sep =</span> <span class="fu">c</span>(<span class="st">":"</span>, <span class="st">"-"</span>), </span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">fragments =</span> <span class="fu">paste0</span>(<span class="st">"data/"</span>, sampleID, <span class="st">"/fragments.tsv.gz"</span>),</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">min.cells =</span> <span class="dv">10</span>, <span class="at">min.features =</span> <span class="dv">200</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> chrom_assay, <span class="at">assay =</span> <span class="st">"peaks"</span>, <span class="at">meta.data =</span> metadata_df)</span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>peaks_keep <span class="ot">=</span> <span class="fu">seqnames</span>(<span class="fu">granges</span>(BM_WT)) <span class="sc">%in%</span> <span class="fu">standardChromosomes</span>(<span class="fu">granges</span>(BM_WT))</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> BM_WT[<span class="fu">as.vector</span>(peaks_keep), ]</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Annotation</span>(BM_WT) <span class="ot">=</span> annotations</span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>BM_WT<span class="sc">$</span>dataset <span class="ot">=</span> sampleID</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">NucleosomeSignal</span>(<span class="at">object =</span> BM_WT)</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">TSSEnrichment</span>(<span class="at">object =</span> BM_WT, <span class="at">fast =</span> T)</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>BM_WT<span class="sc">$</span>blacklist_ratio <span class="ot">=</span> BM_WT<span class="sc">$</span>blacklist_region_fragments <span class="sc">/</span> BM_WT<span class="sc">$</span>peak_region_fragments</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>BM_WT<span class="sc">$</span>pct_reads_in_peaks <span class="ot">=</span> BM_WT<span class="sc">$</span>peak_region_fragments <span class="sc">/</span> BM_WT<span class="sc">$</span>passed_filters <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>BM_WT<span class="sc">$</span>percent.mt <span class="ot">=</span> BM_WT<span class="sc">$</span>mitochondrial <span class="sc">/</span> BM_WT<span class="sc">$</span>passed_filters <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">subset</span>(BM_WT, percent.mt <span class="sc">&lt;</span> <span class="dv">25</span>)</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a>low_prf <span class="ot">=</span> <span class="fu">quantile</span>(BM_WT<span class="sc">$</span>peak_region_fragments, <span class="at">probs =</span> <span class="fl">0.02</span>)</span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>hig_prf <span class="ot">=</span> <span class="fu">quantile</span>(BM_WT<span class="sc">$</span>peak_region_fragments, <span class="at">probs =</span> <span class="fl">0.995</span>)</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>low_prp <span class="ot">=</span> <span class="fu">quantile</span>(BM_WT<span class="sc">$</span>pct_reads_in_peaks, <span class="at">probs =</span> <span class="fl">0.02</span>)</span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>hig_ns <span class="ot">=</span> <span class="fu">quantile</span>(BM_WT<span class="sc">$</span>nucleosome_signal, <span class="at">probs =</span> <span class="fl">0.995</span>)</span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>low_ts <span class="ot">=</span> <span class="fu">quantile</span>(BM_WT<span class="sc">$</span>TSS.enrichment, <span class="at">probs =</span> <span class="fl">0.02</span>)</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>scEX_SampleObject <span class="ot">=</span> <span class="fu">as.SingleCellExperiment</span>(BM_WT)</span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>scEX_SampleObject <span class="ot">=</span> <span class="fu">scDblFinder</span>(scEX_SampleObject, </span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>                                <span class="at">aggregateFeatures=</span><span class="cn">TRUE</span>, </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>                                <span class="at">nfeatures=</span><span class="dv">25</span>, </span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>                                <span class="at">processing=</span><span class="st">"normFeatures"</span>)</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>class <span class="ot">=</span> scEX_SampleObject<span class="sc">$</span>scDblFinder.class; BM_WT<span class="sc">$</span>Class <span class="ot">=</span> class</span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>score <span class="ot">=</span> scEX_SampleObject<span class="sc">$</span>scDblFinder.score; BM_WT<span class="sc">$</span>Score <span class="ot">=</span> score</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>weight <span class="ot">=</span> scEX_SampleObject<span class="sc">$</span>scDblFinder.weighted; BM_WT<span class="sc">$</span>Weight <span class="ot">=</span> weight</span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">subset</span>(BM_WT, <span class="at">subset =</span> Class <span class="sc">==</span> <span class="st">"singlet"</span>)</span>
<span id="cb20-40"><a href="#cb20-40" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">subset</span>(<span class="at">x =</span> BM_WT, </span>
<span id="cb20-41"><a href="#cb20-41" aria-hidden="true" tabindex="-1"></a>               <span class="at">subset =</span> peak_region_fragments <span class="sc">&gt;</span> low_prf <span class="sc">&amp;</span> peak_region_fragments <span class="sc">&lt;</span> hig_prf <span class="sc">&amp;</span></span>
<span id="cb20-42"><a href="#cb20-42" aria-hidden="true" tabindex="-1"></a>                        pct_reads_in_peaks <span class="sc">&gt;</span> low_prp <span class="sc">&amp;</span></span>
<span id="cb20-43"><a href="#cb20-43" aria-hidden="true" tabindex="-1"></a>                        blacklist_ratio <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span></span>
<span id="cb20-44"><a href="#cb20-44" aria-hidden="true" tabindex="-1"></a>                        nucleosome_signal <span class="sc">&lt;</span> hig_ns <span class="sc">&amp;</span></span>
<span id="cb20-45"><a href="#cb20-45" aria-hidden="true" tabindex="-1"></a>                        TSS.enrichment <span class="sc">&gt;</span> low_ts)</span>
<span id="cb20-46"><a href="#cb20-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-47"><a href="#cb20-47" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">=</span> <span class="fu">CallPeaks</span>(<span class="at">object =</span> BM_WT, <span class="at">group.by =</span> <span class="st">"orig.ident"</span>,</span>
<span id="cb20-48"><a href="#cb20-48" aria-hidden="true" tabindex="-1"></a>                  <span class="at">macs2.path =</span> <span class="st">"/opt/miniconda3/envs/scrna_env/bin/macs3"</span>, </span>
<span id="cb20-49"><a href="#cb20-49" aria-hidden="true" tabindex="-1"></a>                  <span class="at">additional.args =</span> <span class="st">"-f BEDPE -g 2.6e9 -q 0.2"</span>)</span>
<span id="cb20-50"><a href="#cb20-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-51"><a href="#cb20-51" aria-hidden="true" tabindex="-1"></a>selected_chromosomes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span>, <span class="st">"chr4"</span>, <span class="st">"chr5"</span>, <span class="st">"chr6"</span>, <span class="st">"chr7"</span>, </span>
<span id="cb20-52"><a href="#cb20-52" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"chr8"</span>, <span class="st">"chr9"</span>, <span class="st">"chr10"</span>, <span class="st">"chr11"</span>, <span class="st">"chr12"</span>, <span class="st">"chr13"</span>, </span>
<span id="cb20-53"><a href="#cb20-53" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"chr14"</span>, <span class="st">"chr15"</span>, <span class="st">"chr16"</span>, <span class="st">"chr17"</span>, <span class="st">"chr18"</span>, <span class="st">"chr19"</span>, </span>
<span id="cb20-54"><a href="#cb20-54" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"chrX"</span>, <span class="st">"chrY"</span>, <span class="st">"chrMT"</span>)</span>
<span id="cb20-55"><a href="#cb20-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-56"><a href="#cb20-56" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">&lt;-</span> <span class="fu">subset</span>(peaks, seqnames <span class="sc">%in%</span> selected_chromosomes)</span>
<span id="cb20-57"><a href="#cb20-57" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">seqlevelsInUse</span>(peaks)</span>
<span id="cb20-58"><a href="#cb20-58" aria-hidden="true" tabindex="-1"></a><span class="fu">seqnames</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(<span class="fu">seqnames</span>(peaks))</span>
<span id="cb20-59"><a href="#cb20-59" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(peaks, <span class="at">file =</span> <span class="fu">paste0</span>(output_dir, sampleID, <span class="st">"_macs3_peaks.rds"</span>))</span>
<span id="cb20-60"><a href="#cb20-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-61"><a href="#cb20-61" aria-hidden="true" tabindex="-1"></a>feature_matrix <span class="ot">&lt;-</span> <span class="fu">FeatureMatrix</span>(<span class="at">fragments =</span> <span class="fu">Fragments</span>(BM_WT), </span>
<span id="cb20-62"><a href="#cb20-62" aria-hidden="true" tabindex="-1"></a>                                <span class="at">features =</span> peaks)</span>
<span id="cb20-63"><a href="#cb20-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-64"><a href="#cb20-64" aria-hidden="true" tabindex="-1"></a>chrom_assay <span class="ot">&lt;-</span> <span class="fu">CreateChromatinAssay</span>(<span class="at">counts =</span> feature_matrix, </span>
<span id="cb20-65"><a href="#cb20-65" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">fragments =</span> <span class="fu">Fragments</span>(BM_WT), </span>
<span id="cb20-66"><a href="#cb20-66" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">annotation =</span> <span class="fu">Annotation</span>(BM_WT))</span>
<span id="cb20-67"><a href="#cb20-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-68"><a href="#cb20-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Add to Seurat object</span></span>
<span id="cb20-69"><a href="#cb20-69" aria-hidden="true" tabindex="-1"></a>BM_WT[[<span class="st">"peaks"</span>]] <span class="ot">&lt;-</span> chrom_assay</span>
<span id="cb20-70"><a href="#cb20-70" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">FindTopFeatures</span>(BM_WT, <span class="at">min.cutoff =</span> <span class="st">"q60"</span>)</span>
<span id="cb20-71"><a href="#cb20-71" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">RunTFIDF</span>(BM_WT)</span>
<span id="cb20-72"><a href="#cb20-72" aria-hidden="true" tabindex="-1"></a>BM_WT <span class="ot">=</span> <span class="fu">RunSVD</span>(BM_WT)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>sampleID <span class="ot">=</span> <span class="st">"BM_KO"</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>counts_mat <span class="ot">=</span> <span class="fu">Read10X_h5</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(<span class="st">"data/"</span>, sampleID, <span class="st">"/filtered_peak_bc_matrix.h5"</span>))</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>metadata_df <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"data/"</span>, sampleID, <span class="st">"/singlecell.csv"</span>), </span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>chrom_assay <span class="ot">=</span> <span class="fu">CreateChromatinAssay</span>(<span class="at">counts =</span> counts_mat, <span class="at">sep =</span> <span class="fu">c</span>(<span class="st">":"</span>, <span class="st">"-"</span>), </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">fragments =</span> <span class="fu">paste0</span>(<span class="st">"data/"</span>, sampleID, <span class="st">"/fragments.tsv.gz"</span>),</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">min.cells =</span> <span class="dv">10</span>, <span class="at">min.features =</span> <span class="dv">200</span>)</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> chrom_assay, <span class="at">assay =</span> <span class="st">"peaks"</span>, <span class="at">meta.data =</span> metadata_df)</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>peaks_keep <span class="ot">=</span> <span class="fu">seqnames</span>(<span class="fu">granges</span>(BM_KO)) <span class="sc">%in%</span> <span class="fu">standardChromosomes</span>(<span class="fu">granges</span>(BM_KO))</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> BM_KO[<span class="fu">as.vector</span>(peaks_keep), ]</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="fu">Annotation</span>(BM_KO) <span class="ot">=</span> annotations</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>BM_KO<span class="sc">$</span>dataset <span class="ot">=</span> sampleID</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">NucleosomeSignal</span>(<span class="at">object =</span> BM_KO)</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">TSSEnrichment</span>(<span class="at">object =</span> BM_KO, <span class="at">fast =</span> T)</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>BM_KO<span class="sc">$</span>blacklist_ratio <span class="ot">=</span> BM_KO<span class="sc">$</span>blacklist_region_fragments <span class="sc">/</span> BM_KO<span class="sc">$</span>peak_region_fragments</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>BM_KO<span class="sc">$</span>pct_reads_in_peaks <span class="ot">=</span> BM_KO<span class="sc">$</span>peak_region_fragments <span class="sc">/</span> BM_KO<span class="sc">$</span>passed_filters <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>BM_KO<span class="sc">$</span>percent.mt <span class="ot">=</span> BM_KO<span class="sc">$</span>mitochondrial <span class="sc">/</span> BM_KO<span class="sc">$</span>passed_filters <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">subset</span>(BM_KO, percent.mt <span class="sc">&lt;</span> <span class="dv">25</span>)</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>low_prf <span class="ot">=</span> <span class="fu">quantile</span>(BM_KO<span class="sc">$</span>peak_region_fragments, <span class="at">probs =</span> <span class="fl">0.02</span>)</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>hig_prf <span class="ot">=</span> <span class="fu">quantile</span>(BM_KO<span class="sc">$</span>peak_region_fragments, <span class="at">probs =</span> <span class="fl">0.995</span>)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>low_prp <span class="ot">=</span> <span class="fu">quantile</span>(BM_KO<span class="sc">$</span>pct_reads_in_peaks, <span class="at">probs =</span> <span class="fl">0.02</span>)</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>hig_ns <span class="ot">=</span> <span class="fu">quantile</span>(BM_KO<span class="sc">$</span>nucleosome_signal, <span class="at">probs =</span> <span class="fl">0.995</span>)</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>low_ts <span class="ot">=</span> <span class="fu">quantile</span>(BM_KO<span class="sc">$</span>TSS.enrichment, <span class="at">probs =</span> <span class="fl">0.02</span>)</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>scEX_SampleObject <span class="ot">=</span> <span class="fu">as.SingleCellExperiment</span>(BM_KO)</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>scEX_SampleObject <span class="ot">=</span> <span class="fu">scDblFinder</span>(scEX_SampleObject, </span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>                                <span class="at">aggregateFeatures=</span><span class="cn">TRUE</span>, </span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>                                <span class="at">nfeatures=</span><span class="dv">25</span>, </span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>                                <span class="at">processing=</span><span class="st">"normFeatures"</span>)</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>class <span class="ot">=</span> scEX_SampleObject<span class="sc">$</span>scDblFinder.class; BM_KO<span class="sc">$</span>Class <span class="ot">=</span> class</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>score <span class="ot">=</span> scEX_SampleObject<span class="sc">$</span>scDblFinder.score; BM_KO<span class="sc">$</span>Score <span class="ot">=</span> score</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>weight <span class="ot">=</span> scEX_SampleObject<span class="sc">$</span>scDblFinder.weighted; BM_KO<span class="sc">$</span>Weight <span class="ot">=</span> weight</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">subset</span>(BM_KO, <span class="at">subset =</span> Class <span class="sc">==</span> <span class="st">"singlet"</span>)</span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">subset</span>(<span class="at">x =</span> BM_KO, </span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>               <span class="at">subset =</span> peak_region_fragments <span class="sc">&gt;</span> low_prf <span class="sc">&amp;</span> peak_region_fragments <span class="sc">&lt;</span> hig_prf <span class="sc">&amp;</span></span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>                        pct_reads_in_peaks <span class="sc">&gt;</span> low_prp <span class="sc">&amp;</span></span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a>                        blacklist_ratio <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span></span>
<span id="cb21-44"><a href="#cb21-44" aria-hidden="true" tabindex="-1"></a>                        nucleosome_signal <span class="sc">&lt;</span> hig_ns <span class="sc">&amp;</span></span>
<span id="cb21-45"><a href="#cb21-45" aria-hidden="true" tabindex="-1"></a>                        TSS.enrichment <span class="sc">&gt;</span> low_ts)</span>
<span id="cb21-46"><a href="#cb21-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-47"><a href="#cb21-47" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">=</span> <span class="fu">CallPeaks</span>(<span class="at">object =</span> BM_KO, <span class="at">group.by =</span> <span class="st">"orig.ident"</span>,</span>
<span id="cb21-48"><a href="#cb21-48" aria-hidden="true" tabindex="-1"></a>                  <span class="at">macs2.path =</span> <span class="st">"/opt/miniconda3/envs/scrna_env/bin/macs3"</span>, </span>
<span id="cb21-49"><a href="#cb21-49" aria-hidden="true" tabindex="-1"></a>                  <span class="at">additional.args =</span> <span class="st">"-f BEDPE -g 2.6e9 -q 0.2"</span>)</span>
<span id="cb21-50"><a href="#cb21-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-51"><a href="#cb21-51" aria-hidden="true" tabindex="-1"></a>selected_chromosomes <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span>, <span class="st">"chr4"</span>, <span class="st">"chr5"</span>, <span class="st">"chr6"</span>, <span class="st">"chr7"</span>, </span>
<span id="cb21-52"><a href="#cb21-52" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"chr8"</span>, <span class="st">"chr9"</span>, <span class="st">"chr10"</span>, <span class="st">"chr11"</span>, <span class="st">"chr12"</span>, <span class="st">"chr13"</span>, </span>
<span id="cb21-53"><a href="#cb21-53" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"chr14"</span>, <span class="st">"chr15"</span>, <span class="st">"chr16"</span>, <span class="st">"chr17"</span>, <span class="st">"chr18"</span>, <span class="st">"chr19"</span>, </span>
<span id="cb21-54"><a href="#cb21-54" aria-hidden="true" tabindex="-1"></a>                          <span class="st">"chrX"</span>, <span class="st">"chrY"</span>, <span class="st">"chrMT"</span>)</span>
<span id="cb21-55"><a href="#cb21-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-56"><a href="#cb21-56" aria-hidden="true" tabindex="-1"></a>peaks <span class="ot">&lt;-</span> <span class="fu">subset</span>(peaks, seqnames <span class="sc">%in%</span> selected_chromosomes)</span>
<span id="cb21-57"><a href="#cb21-57" aria-hidden="true" tabindex="-1"></a><span class="fu">seqlevels</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">seqlevelsInUse</span>(peaks)</span>
<span id="cb21-58"><a href="#cb21-58" aria-hidden="true" tabindex="-1"></a><span class="fu">seqnames</span>(peaks) <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(<span class="fu">seqnames</span>(peaks))</span>
<span id="cb21-59"><a href="#cb21-59" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(peaks, <span class="at">file =</span> <span class="fu">paste0</span>(output_dir, sampleID, <span class="st">"_macs3_peaks.rds"</span>))</span>
<span id="cb21-60"><a href="#cb21-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-61"><a href="#cb21-61" aria-hidden="true" tabindex="-1"></a>feature_matrix <span class="ot">&lt;-</span> <span class="fu">FeatureMatrix</span>(<span class="at">fragments =</span> <span class="fu">Fragments</span>(BM_KO), </span>
<span id="cb21-62"><a href="#cb21-62" aria-hidden="true" tabindex="-1"></a>                                <span class="at">features =</span> peaks)</span>
<span id="cb21-63"><a href="#cb21-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-64"><a href="#cb21-64" aria-hidden="true" tabindex="-1"></a>chrom_assay <span class="ot">&lt;-</span> <span class="fu">CreateChromatinAssay</span>(<span class="at">counts =</span> feature_matrix, </span>
<span id="cb21-65"><a href="#cb21-65" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">fragments =</span> <span class="fu">Fragments</span>(BM_KO), </span>
<span id="cb21-66"><a href="#cb21-66" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">annotation =</span> <span class="fu">Annotation</span>(BM_KO))</span>
<span id="cb21-67"><a href="#cb21-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-68"><a href="#cb21-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Add to Seurat object</span></span>
<span id="cb21-69"><a href="#cb21-69" aria-hidden="true" tabindex="-1"></a>BM_KO[[<span class="st">"peaks"</span>]] <span class="ot">&lt;-</span> chrom_assay</span>
<span id="cb21-70"><a href="#cb21-70" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">FindTopFeatures</span>(BM_KO, <span class="at">min.cutoff =</span> <span class="st">"q60"</span>)</span>
<span id="cb21-71"><a href="#cb21-71" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">RunTFIDF</span>(BM_KO)</span>
<span id="cb21-72"><a href="#cb21-72" aria-hidden="true" tabindex="-1"></a>BM_KO <span class="ot">=</span> <span class="fu">RunSVD</span>(BM_KO)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">merge</span>(BM_WT, BM_KO)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">FindTopFeatures</span>(joint_atac, <span class="at">min.cutoff =</span> <span class="st">"q60"</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">RunTFIDF</span>(joint_atac)</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">RunSVD</span>(joint_atac)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">RunUMAP</span>(joint_atac, <span class="at">reduction =</span> <span class="st">'lsi'</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">FindNeighbors</span>(joint_atac, <span class="at">reduction =</span> <span class="st">'lsi'</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">FindClusters</span>(joint_atac, <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">algorithm =</span> <span class="dv">3</span>, <span class="at">resolution =</span> <span class="fl">0.6</span>)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a><span class="fu">Idents</span>(joint_atac) <span class="ot">=</span> <span class="st">"dataset"</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>ranges <span class="ot">&lt;-</span> joint_atac<span class="sc">@</span>assays<span class="sc">$</span>peaks<span class="sc">@</span>ranges</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="fu">seqinfo</span>(ranges) <span class="ot">&lt;-</span> <span class="fu">seqinfo</span>(BSgenome.Mmusculus.UCSC.mm10)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>joint_atac<span class="sc">@</span>assays<span class="sc">$</span>peaks<span class="sc">@</span>ranges <span class="ot">&lt;-</span> ranges</span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>harmony_embeddings <span class="ot">=</span> harmony<span class="sc">::</span><span class="fu">RunHarmony</span>(joint_atac<span class="sc">@</span>reductions<span class="sc">$</span>lsi<span class="sc">@</span>cell.embeddings, joint_atac<span class="sc">@</span>meta.data<span class="sc">$</span>dataset)</span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>joint_atac[[<span class="st">"harmony"</span>]] <span class="ot">=</span> harmony_embeddings</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>joint_atac<span class="sc">@</span>reductions[[<span class="st">'harmony'</span>]] <span class="ot">=</span> <span class="fu">CreateDimReducObject</span>(<span class="at">embeddings =</span> harmony_embeddings, <span class="at">key =</span> <span class="st">'harmony_'</span>, <span class="at">assay =</span> <span class="st">'peak'</span>)</span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">RunUMAP</span>(joint_atac, <span class="at">reduction =</span> <span class="st">'harmony'</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">FindNeighbors</span>(joint_atac, <span class="at">reduction =</span> <span class="st">'harmony'</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>, <span class="at">graph.name =</span> <span class="st">"harmony_graph"</span>)</span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">FindClusters</span>(joint_atac, <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">algorithm =</span> <span class="dv">3</span>, <span class="at">resolution =</span> <span class="fl">0.8</span>, <span class="at">graph.name =</span> <span class="st">"harmony_graph"</span>, <span class="at">cluster.name =</span> <span class="st">"harmony_clusters"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>gene_activities <span class="ot">=</span> <span class="fu">GeneActivity</span>(joint_atac)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>joint_atac[[<span class="st">"RNA"</span>]] <span class="ot">=</span> <span class="fu">CreateAssayObject</span>(<span class="at">counts =</span> gene_activities)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">NormalizeData</span>(<span class="at">object =</span> joint_atac, </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">assay =</span> <span class="st">'RNA'</span>, </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">normalization.method =</span> <span class="st">'LogNormalize'</span>, </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">scale.factor =</span> <span class="fu">median</span>(joint_atac<span class="sc">$</span>nCount_RNA))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>BM_rna <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"Hegde_2025_Mouse_BM_Nfe2l2_scRNA.rds"</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>transfer.anchors <span class="ot">&lt;-</span> <span class="fu">FindTransferAnchors</span>(</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">reference =</span> BM_rna, </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">query =</span> joint_atac,</span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">'cca'</span>,</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>predicted.labels <span class="ot">&lt;-</span> <span class="fu">TransferData</span>(</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">anchorset =</span> transfer.anchors,</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">refdata =</span> BM_rna<span class="sc">$</span>annots,</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight.reduction =</span> joint_atac[[<span class="st">'harmony'</span>]],</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">AddMetaData</span>(<span class="at">object =</span> joint_atac, <span class="at">metadata =</span> predicted.labels)</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Remove noisy clusters ----</span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">subset</span>(joint_atac, harmony_clusters <span class="sc">%notin%</span> <span class="fu">c</span>(<span class="dv">10</span>, <span class="dv">13</span>, <span class="dv">15</span>, <span class="dv">17</span>, <span class="dv">19</span>, <span class="dv">20</span>, <span class="dv">21</span>, <span class="dv">22</span>))</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">RunUMAP</span>(joint_atac, <span class="at">reduction =</span> <span class="st">'harmony'</span>, <span class="at">dims =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(JASPAR2024)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>jaspar <span class="ot">=</span> <span class="fu">JASPAR2024</span>()</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>sq24 <span class="ot">=</span> RSQLite<span class="sc">::</span><span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), <span class="fu">db</span>(jaspar))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>pwms <span class="ot">=</span> TFBSTools<span class="sc">::</span><span class="fu">getMatrixSet</span>(sq24, <span class="fu">list</span>(<span class="at">species =</span> <span class="st">"Mus musculus"</span>, <span class="at">matrixtype =</span> <span class="st">"PWM"</span>, <span class="at">collection =</span> <span class="st">"CORE"</span>))</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>sq24 <span class="ot">=</span> RSQLite<span class="sc">::</span><span class="fu">dbConnect</span>(RSQLite<span class="sc">::</span><span class="fu">SQLite</span>(), <span class="fu">db</span>(jaspar))</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">AddMotifs</span>(<span class="at">object =</span> joint_atac, </span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                       <span class="at">genome =</span> BSgenome.Mmusculus.UCSC.mm10, </span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">pfm =</span> pwms)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">&lt;-</span> <span class="fu">RunChromVAR</span>(<span class="at">object =</span> joint_atac, <span class="at">genome =</span> BSgenome.Mmusculus.UCSC.mm10)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>joint_atac <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">"signac/Hegde_2025_Mouse_BM_Nfe2l2_scATAC.rds"</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(joint_atac<span class="sc">$</span>annotated, joint_atac<span class="sc">$</span>harmony_clusters)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    
                        0    1   10   11   12   13   14   15   16   17   18
  Baso                  0    0    0    0    0    0  287    0    0    0    0
  cMoP                 33    7    0    0    0    0    0    0    0    0    0
  GMP                   0   38    0    0    0    0    0    0    2    0  137
  GP                    0    8    0    0    0    0    0    0    0    0    0
  Ly6Chi mono        1101    0    0  174    0    0    0    0    0    0    0
  Ly6Chi proDC3         4    0    0  280    2    0    0    0    0    0    0
  Ly6Clow mono          0    0    0    7  421    0    0    0    8    0    0
  MEP                   0    0    0    0    0    0    2    0  178    0    0
  Neu_immature          0  225    0    0    0    0    1    0    0    0    0
  Neu_mature            0    6    0    0    0    0    0    0    0    0    0
  Neu_preNeu            2  814    0    1    1    0    3    0    4    0    0
  Neu_preNeu_cycling    0   35    0    0    0    0    0    0    1    0    0
  Neu_proNeu            0    9    0    0    0    0    0    0    0    0    0
  Neu_proNeu_cycling    0   14    0    0    0    0    0    0    2    0    0
  pre-mono            296    1    0    2    0    0    0    0    3    0    0
                    
                       19    2   20   21   22    3    4    5    6    7    8
  Baso                  0    0    0    0    0    0    0    0    0    0    0
  cMoP                  0  555    0    0    0    1  777    0    0    0    0
  GMP                   0  158    0    0    0    6    1    0    0    0   75
  GP                    0    0    0    0    0    7    0    5    0   32  314
  Ly6Chi mono           0   24    0    0    0    0   12    0    0    0    0
  Ly6Chi proDC3         0    3    0    0    0    0    0    0    0    0    0
  Ly6Clow mono          0    2    0    0    0    0    0    0    0    0    0
  MEP                   0    0    0    0    0    0    0    0    0    0    0
  Neu_immature          0    9    0    0    0    0    1    0   52    0   20
  Neu_mature            0    4    0    0    0    0    0    0  726    0    3
  Neu_preNeu            0  184    0    0    0   51    0    3    0   28  119
  Neu_preNeu_cycling    0    0    0    0    0  676    0  401    0   40   21
  Neu_proNeu            0    0    0    0    0  113    0   13    0  298    2
  Neu_proNeu_cycling    0    0    0    0    0   13    0  425    0  186    5
  pre-mono              0   23    0    0    0    1   71    0    0    0    0
                    
                        9
  Baso                  0
  cMoP                 41
  GMP                 207
  GP                    0
  Ly6Chi mono           7
  Ly6Chi proDC3       224
  Ly6Clow mono          0
  MEP                   0
  Neu_immature          0
  Neu_mature            0
  Neu_preNeu           19
  Neu_preNeu_cycling    0
  Neu_proNeu            0
  Neu_proNeu_cycling    0
  pre-mono             13</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>SCP<span class="sc">::</span><span class="fu">CellDimPlot</span>(joint_atac, <span class="at">group.by =</span> <span class="st">"annotated"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: No shared levels found between `names(values)` of the manual scale and the
data's fill values.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>annot_vec <span class="ot">=</span> <span class="fu">unique</span>(joint_atac<span class="sc">$</span>annotated)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>holder1 <span class="ot">=</span> <span class="fu">tibble</span>()</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(annot_vec)) {</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>  iter_annot <span class="ot">=</span> annot_vec[i]; <span class="fu">print</span>(iter_annot)</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>  atac_ss <span class="ot">=</span> <span class="fu">subset</span>(joint_atac, annotated <span class="sc">==</span> iter_annot)</span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">DefaultAssay</span>(atac_ss) <span class="ot">=</span> <span class="st">"peaks"</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">Idents</span>(atac_ss) <span class="ot">=</span> <span class="st">"dataset"</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  atac_ss<span class="sc">$</span>dataset <span class="ot">=</span> <span class="fu">factor</span>(atac_ss<span class="sc">$</span>dataset, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"BM_WT"</span>, <span class="st">"BM_KO"</span>))</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a>  da_peaks <span class="ot">=</span> presto<span class="sc">::</span><span class="fu">wilcoxauc</span>(atac_ss<span class="sc">@</span>assays<span class="sc">$</span>peaks<span class="sc">$</span>data, <span class="at">y =</span> atac_ss<span class="sc">$</span>dataset)</span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>  da_peaks<span class="sc">$</span>closest_gene <span class="ot">=</span> <span class="fu">ClosestFeature</span>(atac_ss, <span class="at">regions =</span> da_peaks<span class="sc">$</span>feature)<span class="sc">$</span>gene_name</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a>  da_peaks<span class="sc">$</span>distance <span class="ot">=</span> <span class="fu">ClosestFeature</span>(atac_ss, <span class="at">regions =</span> da_peaks<span class="sc">$</span>feature)<span class="sc">$</span>distance</span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  da_peaks <span class="ot">=</span> da_peaks <span class="sc">|&gt;</span> </span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> <span class="st">"BM_KO"</span>) <span class="sc">|&gt;</span></span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">separate</span>(feature, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>), <span class="at">sep =</span> <span class="st">"-"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">start =</span> <span class="fu">as.numeric</span>(start), <span class="at">end =</span> <span class="fu">as.numeric</span>(end), </span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">ident_name =</span> iter_annot,</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>           <span class="at">pct_diff =</span> pct_in <span class="sc">-</span> pct_out,</span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>           <span class="at">peakID =</span> <span class="fu">paste</span>(seqnames, start, end, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>           <span class="at">group_test =</span> iter_annot)</span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  da_peaks <span class="ot">&lt;-</span> da_peaks[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^Gm|^RP|Rik$|Erdr1|Mid1|Ccr1|Rik8$|^a$"</span>, da_peaks<span class="sc">$</span>closest_gene),]</span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>  da_peaks <span class="ot">=</span> da_peaks <span class="sc">|&gt;</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(closest_gene <span class="sc">!=</span> <span class="st">""</span>)</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>  holder1 <span class="ot">=</span> <span class="fu">bind_rows</span>(holder1, da_peaks)</span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "GP"
[1] "cMoP"
[1] "pre-mono"
[1] "Neu_proNeu_cycling"
[1] "Ly6Clow mono"
[1] "Ly6Chi mono"
[1] "GMP"
[1] "Neu_preNeu_cycling"
[1] "Neu_preNeu"
[1] "Neu_proNeu"
[1] "MEP"
[1] "Ly6Chi proDC3"
[1] "Baso"
[1] "Neu_immature"
[1] "Neu_mature"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(holder1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 18
  seqnames   start     end group avgExpr   logFC statistic   auc    pval   padj
  &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;
1 chr1     3445703 3446197 BM_KO   0.154 -0.0439    16139  0.493 0.638   0.847 
2 chr1     4332493 4332694 BM_KO   0     -0.0741    16068  0.490 0.0840  0.361 
3 chr1     4412612 4412767 BM_KO   0.165  0.165     17115  0.522 0.00198 0.0843
4 chr1     4414289 4414479 BM_KO   0.114  0.114     16905  0.516 0.00916 0.153 
5 chr1     4414582 4415044 BM_KO   0.180  0.0747    17002. 0.519 0.131   0.405 
6 chr1     4416545 4417168 BM_KO   0.175 -0.0516    16081  0.491 0.582   0.806 
# ℹ 8 more variables: pct_in &lt;dbl&gt;, pct_out &lt;dbl&gt;, closest_gene &lt;chr&gt;,
#   distance &lt;int&gt;, ident_name &lt;chr&gt;, pct_diff &lt;dbl&gt;, peakID &lt;chr&gt;,
#   group_test &lt;chr&gt;</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>holder1_filtered <span class="ot">=</span> holder1 <span class="sc">|&gt;</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(logFC) <span class="sc">&gt;</span> <span class="fl">0.25</span> <span class="sc">&amp;</span> auc <span class="sc">&gt;</span> <span class="fl">0.3</span>) <span class="sc">|&gt;</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">direction =</span> <span class="fu">case_when</span>(logFC <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"up"</span>,</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>                               logFC <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"down"</span>))</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>summary_df <span class="ot">=</span> holder1_filtered <span class="sc">|&gt;</span> </span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(group_test) <span class="sc">|&gt;</span> </span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(direction, <span class="at">name =</span> <span class="st">"dar_count"</span>) <span class="sc">|&gt;</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(group_test <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Ly6Clow mono"</span>, <span class="st">"Ly6Chi proDC3"</span>, <span class="st">"Ly6Chi mono"</span>,</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"pre-mono"</span>, <span class="st">"cMoP"</span>, <span class="st">"GP"</span>, <span class="st">"GMP"</span>, </span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Neu_proNeu"</span>, <span class="st">"Neu_preNeu"</span>, </span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">"Neu_immature"</span>, <span class="st">"Neu_mature"</span>))</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>summary_df<span class="sc">$</span>group_test <span class="ot">=</span> <span class="fu">factor</span>(summary_df<span class="sc">$</span>group_test, </span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Ly6Clow mono"</span>, <span class="st">"Ly6Chi proDC3"</span>, <span class="st">"Ly6Chi mono"</span>,</span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"pre-mono"</span>, <span class="st">"cMoP"</span>, <span class="st">"GP"</span>, <span class="st">"GMP"</span>, </span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Neu_proNeu"</span>, <span class="st">"Neu_preNeu"</span>, </span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">"Neu_immature"</span>, <span class="st">"Neu_mature"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(summary_df, </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes_string</span>(<span class="at">x =</span> <span class="st">"group_test"</span>, <span class="at">y =</span> <span class="st">"dar_count"</span>, <span class="at">fill =</span> <span class="st">"direction"</span>)) <span class="sc">+</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> dar_count), </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>),</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"up"</span> <span class="ot">=</span> <span class="st">"#050708"</span>, <span class="st">"down"</span> <span class="ot">=</span> <span class="st">"#929391"</span>)) <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_y_continuous</span>(<span class="st">"DAR count"</span>) <span class="sc">+</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>       <span class="fu">scale_x_discrete</span>(<span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme_minimal</span>(<span class="at">base_size =</span> <span class="dv">14</span>) <span class="sc">+</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
ℹ Please use tidy evaluation idioms with `aes()`.
ℹ See also `vignette("ggplot2-in-packages")` for more information.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Hegde_2025_Mouse_BM_Nfe2l2_scATAC_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="export" class="level2">
<h2 class="anchored" data-anchor-id="export">Export</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>joint_atac_bc_annot_df <span class="ot">=</span> joint_atac<span class="sc">@</span>meta.data <span class="sc">|&gt;</span> </span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span> </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">"bc"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(bc, harmony_clusters, annotated)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(joint_atac_bc_annot_df, </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="st">"signac/Hegde_2025_Mouse_BM_Nfe2l2_scATAC_bc_cluster_annot.csv"</span>,</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">row.names =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="session-info" class="level2">
<h2 class="anchored" data-anchor-id="session-info">Session Info</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>end_time <span class="ot">&lt;-</span> <span class="fu">Sys.time</span>()</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(end_time <span class="sc">-</span> start_time)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Time difference of 4.37923 mins</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.4.3 (2025-02-28)
Platform: aarch64-apple-darwin20
Running under: macOS Sequoia 15.4.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] BSgenome.Mmusculus.UCSC.mm10_1.4.3 BSgenome_1.74.0                   
 [3] BiocIO_1.16.0                      Biostrings_2.74.1                 
 [5] XVector_0.46.0                     EnsDb.Mmusculus.v79_2.99.0        
 [7] ensembldb_2.30.0                   AnnotationFilter_1.30.0           
 [9] GenomicFeatures_1.58.0             AnnotationDbi_1.68.0              
[11] TFBSTools_1.44.0                   JASPAR2020_0.99.10                
[13] patchwork_1.3.0                    plyranges_1.26.0                  
[15] rtracklayer_1.66.0                 scDblFinder_1.20.2                
[17] SingleCellExperiment_1.28.1        SummarizedExperiment_1.36.0       
[19] Biobase_2.66.0                     GenomicRanges_1.58.0              
[21] GenomeInfoDb_1.42.3                IRanges_2.40.1                    
[23] S4Vectors_0.44.0                   BiocGenerics_0.52.0               
[25] MatrixGenerics_1.18.1              matrixStats_1.5.0                 
[27] Signac_1.14.0                      SeuratExtend_1.1.6                
[29] SeuratExtendData_0.2.1             Seurat_5.3.0                      
[31] SeuratObject_5.1.99.9000           sp_2.2-0                          
[33] lubridate_1.9.4                    forcats_1.0.0                     
[35] stringr_1.5.1                      dplyr_1.1.4                       
[37] purrr_1.0.4                        readr_2.1.5                       
[39] tidyr_1.3.1                        tibble_3.2.1                      
[41] ggplot2_3.5.2                      tidyverse_2.0.0                   

loaded via a namespace (and not attached):
  [1] R.methodsS3_1.8.2           dichromat_2.0-0.1          
  [3] progress_1.2.3              nnet_7.3-20                
  [5] poweRlaw_1.0.0              goftest_1.2-3              
  [7] HDF5Array_1.34.0            vctrs_0.6.5                
  [9] ggtangle_0.0.6              spatstat.random_3.3-3      
 [11] digest_0.6.37               png_0.1-8                  
 [13] shape_1.4.6.1               slingshot_2.14.0           
 [15] ggrepel_0.9.6               deldir_2.0-4               
 [17] parallelly_1.43.0           MASS_7.3-65                
 [19] reshape2_1.4.4              qvalue_2.38.0              
 [21] httpuv_1.6.16               foreach_1.5.2              
 [23] withr_3.0.2                 ggfun_0.1.8                
 [25] xfun_0.52                   survival_3.8-3             
 [27] proxyC_0.5.2                memoise_2.0.1              
 [29] ggbeeswarm_0.7.2            gson_0.1.0                 
 [31] clusterProfiler_4.14.6      princurve_2.1.6            
 [33] SCP_0.5.6                   tidytree_0.4.6             
 [35] zoo_1.8-14                  GlobalOptions_0.1.2        
 [37] gtools_3.9.5                pbapply_1.7-2              
 [39] R.oo_1.27.0                 prettyunits_1.2.0          
 [41] Formula_1.2-5               KEGGREST_1.46.0            
 [43] promises_1.3.2              httr_1.4.7                 
 [45] restfulr_0.0.15             globals_0.17.0             
 [47] fitdistrplus_1.2-2          rhdf5filters_1.18.1        
 [49] rhdf5_2.50.2                rstudioapi_0.17.1          
 [51] DOSE_4.0.1                  UCSC.utils_1.2.0           
 [53] miniUI_0.1.2                generics_0.1.3             
 [55] base64enc_0.1-3             curl_6.2.2                 
 [57] zlibbioc_1.52.0             ScaledMatrix_1.14.0        
 [59] polyclip_1.10-7             GenomeInfoDbData_1.2.13    
 [61] SparseArray_1.6.2           xtable_1.8-4               
 [63] doParallel_1.0.17           evaluate_1.0.3             
 [65] S4Arrays_1.6.0              BiocFileCache_2.14.0       
 [67] hms_1.1.3                   irlba_2.3.5.1              
 [69] filelock_1.0.3              colorspace_2.1-1           
 [71] ROCR_1.0-11                 reticulate_1.42.0          
 [73] spatstat.data_3.1-6         magrittr_2.0.3             
 [75] lmtest_0.9-40               ggtree_3.14.0              
 [77] later_1.4.2                 viridis_0.6.5              
 [79] lattice_0.22-7              spatstat.geom_3.3-6        
 [81] future.apply_1.11.3         scattermore_1.2            
 [83] XML_3.99-0.18               scuttle_1.16.0             
 [85] cowplot_1.1.3               RcppAnnoy_0.0.22           
 [87] Hmisc_5.2-3                 pillar_1.10.2              
 [89] nlme_3.1-168                iterators_1.0.14           
 [91] pwalign_1.2.0               caTools_1.18.3             
 [93] compiler_4.4.3              beachmat_2.22.0            
 [95] RSpectra_0.16-2             stringi_1.8.7              
 [97] tensor_1.5                  GenomicAlignments_1.42.0   
 [99] plyr_1.8.9                  crayon_1.5.3               
[101] abind_1.4-8                 scater_1.34.1              
[103] gridGraphics_0.5-1          locfit_1.5-9.12            
[105] bit_4.6.0                   fastmatch_1.1-6            
[107] codetools_0.2-20            BiocSingular_1.22.0        
[109] biovizBase_1.54.0           GetoptLong_1.0.5           
[111] plotly_4.10.4               mime_0.13                  
[113] splines_4.4.3               circlize_0.4.16            
[115] Rcpp_1.0.14                 fastDummies_1.7.5          
[117] dbplyr_2.5.0                utf8_1.2.4                 
[119] knitr_1.50                  blob_1.2.4                 
[121] clue_0.3-66                 seqLogo_1.72.0             
[123] fs_1.6.6                    listenv_0.9.1              
[125] checkmate_2.3.2             ggplotify_0.1.2            
[127] Matrix_1.7-3                statmod_1.5.0              
[129] tzdb_0.5.0                  tweenr_2.0.3               
[131] pkgconfig_2.0.3             tools_4.4.3                
[133] cachem_1.1.0                R.cache_0.16.0             
[135] RSQLite_2.3.9               viridisLite_0.4.2          
[137] DBI_1.2.3                   fastmap_1.2.0              
[139] rmarkdown_2.29              scales_1.4.0               
[141] grid_4.4.3                  ica_1.0-3                  
[143] Rsamtools_2.22.0            dotCall64_1.2              
[145] VariantAnnotation_1.52.0    RANN_2.6.2                 
[147] rpart_4.1.24                farver_2.1.2               
[149] yaml_2.3.10                 foreign_0.8-90             
[151] cli_3.6.5                   lifecycle_1.0.4            
[153] uwot_0.2.3                  presto_1.0.0               
[155] bluster_1.16.0              backports_1.5.0            
[157] BiocParallel_1.40.2         annotate_1.84.0            
[159] timechange_0.3.0            gtable_0.3.6               
[161] rjson_0.2.23                ggridges_0.5.6             
[163] progressr_0.15.1            ape_5.8-1                  
[165] parallel_4.4.3              limma_3.62.2               
[167] jsonlite_2.0.0              edgeR_4.4.2                
[169] RcppHNSW_0.6.0              bitops_1.0-9               
[171] bit64_4.6.0-1               xgboost_1.7.10.1           
[173] Rtsne_0.17                  yulab.utils_0.2.0          
[175] spatstat.utils_3.1-3        BiocNeighbors_2.0.1        
[177] CNEr_1.42.0                 metapod_1.14.0             
[179] GOSemSim_2.32.0             dqrng_0.4.1                
[181] spatstat.univar_3.1-2       R.utils_2.13.0             
[183] lazyeval_0.2.2              shiny_1.10.0               
[185] enrichplot_1.26.6           htmltools_0.5.8.1          
[187] GO.db_3.20.0                sctransform_0.4.1          
[189] rappdirs_0.3.3              glue_1.8.0                 
[191] TFMPvalue_0.0.9             spam_2.11-1                
[193] httr2_1.1.2                 RCurl_1.98-1.17            
[195] treeio_1.30.0               scran_1.34.0               
[197] gridExtra_2.3               TrajectoryUtils_1.14.0     
[199] igraph_2.1.4                R6_2.6.1                   
[201] labeling_0.4.3              RcppRoll_0.3.1             
[203] cluster_2.1.8.1             Rhdf5lib_1.28.0            
[205] aplot_0.2.5                 DirichletMultinomial_1.48.0
[207] DelayedArray_0.32.0         tidyselect_1.2.1           
[209] vipor_0.4.7                 ProtGenerics_1.38.0        
[211] htmlTable_2.4.3             ggforce_0.4.2              
[213] xml2_1.3.8                  future_1.40.0              
[215] rsvd_1.0.5                  KernSmooth_2.23-26         
[217] data.table_1.17.0           fgsea_1.32.4               
[219] htmlwidgets_1.6.4           ComplexHeatmap_2.22.0      
[221] RColorBrewer_1.1-3          biomaRt_2.62.1             
[223] rlang_1.1.6                 spatstat.sparse_3.1-0      
[225] spatstat.explore_3.4-2      ggnewscale_0.5.1           
[227] beeswarm_0.4.0             </code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>